pkg/scheduler/core/generic_scheduler.go

```diff
@@ -253,23 +253,36 @@ func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes i
 // Filters the nodes to find the ones that fit the pod based on the framework
 // filter plugins and filter extenders.
 func (g *genericScheduler) findNodesThatFitPod(ctx context.Context, prof *profile.Profile, state *framework.CycleState, pod *v1.Pod) ([]*v1.Node, framework.NodeToStatusMap, error) {
+	filteredNodesStatuses := make(framework.NodeToStatusMap)
+
 	// Run "prefilter" plugins.
 	s := prof.RunPreFilterPlugins(ctx, state, pod)
 	if !s.IsSuccess() {
+		if s.IsUnschedulable() {
+			// All nodes will have the same status. Some non trivial refactoring is
+			// needed to avoid this copy.
+			allNodes, err := g.nodeInfoSnapshot.NodeInfos().List()
+			if err != nil {
+				return nil, nil, err
+			}
+			for _, n := range allNodes {
+				filteredNodesStatuses[n.Node().Name] = s
+			}
+			return nil, filteredNodesStatuses, nil
+		}
```

Comment:
nit, in terms of "indenting negative path":

```go
	if !s.IsSuccess() {
		if !s.IsUnschedulable() {
			return nil, nil, s.AsError()
		}
		// All nodes will have the same status. Some non trivial refactoring is
		// needed to avoid this copy.
		allNodes, err := g.nodeInfoSnapshot.NodeInfos().List()
		if err != nil {
			return nil, nil, err
		}
		for _, n := range allNodes {
			filteredNodesStatuses[n.Node().Name] = s
		}
		return nil, filteredNodesStatuses, nil
	}
```

Reply:
done